src/widgets/table/table.rs

```diff
@@ -1197,6 +1197,74 @@ mod tests {
             ]);
             assert_eq!(buf, expected);
         }
+
+        /// Regression test for <https://github.com/ratatui-org/ratatui/issues/1179>
+        ///
+        /// A bug where the table would not render the correct rows when there is no selection.
+        #[test]
+        fn render_with_offset() {
+            let rows = (1..100).map(|i| Row::new([i.to_string()]));
+            let table = Table::new(rows, [Constraint::Length(2)]);
+            let mut buf = Buffer::empty(Rect::new(0, 0, 2, 5));
+            let mut state = TableState::new().with_offset(50);
+            StatefulWidget::render(table, Rect::new(0, 0, 5, 5), &mut buf, &mut state);
+            assert_eq!(buf, Buffer::with_lines(["51", "52", "53", "54", "55"]));
+        }
+
+        #[test]
+        fn render_with_selection_and_offset() {
```

Comment:
This looks like a good rstest candidate.

Reply:
Agreed 3482bbca3

Reply:
Make that 62320257d (the no selection case can be handled in the same test)

Reply:
The rstest is way easier to read :sparkles: